### PR TITLE
CLOS-OSPF Persistent Configuration

### PR DESCRIPTION
**Pull request by slaffer-au**
Currently the Cumulus VX guests which are deployed in these demos lose their configuration in the event they are reloaded. This is due to the provisioning programming the kernel and Quagga directly, rather than writing to persistent configuration files.

This branch updates the Ansible provisioning scripts for the CLOS-OSFP demo to write configuration directly to /etc/network/interfaces* and /etc/quagga/Quagga.conf.

One side note is that is does rely on the cl-interfaces Ansible module being installed. We can add that as a step to the VX+Vagrant User Guide until the module is bundled in an Ansible release.

### CODE REVIEW

File: vagrant/demos/clos-ospf/Vagrantfile [hunk 1]
```diff
@@ -32,7 +34,14 @@ Vagrant.configure("2") do |config|
...
           loopback_ip: properties[:ipAddrPrefix] + iter.to_s
         }
-        ansible.playbook = "clospf.yml"
+        ansible.playbook = "provisioning/clospf.yml"
+      end
+      
+      node.vm.provision "shell" do |s|
+        ssh_pub_key = File.readlines("#{Dir.home}/.ssh/id_rsa.pub").first.strip
+        s.inline = <<-SHELL
+            echo #{ssh_pub_key} >> /home/vagrant/.ssh/authorized_keys
```
Comment (Vanders) [inline vagrant/demos/clos-ospf/Vagrantfile]: I'm guessing you're adding your own SSH pubkey do the Vagrant use so can log in directly with an SSH client? I'm not sure that's robust; we should stick to "vagrant ssh" to have Vagrant handle it all for you by logging in with its own key.

File: vagrant/demos/clos-ospf/Vagrantfile [hunk 2]
```diff
@@ -55,9 +64,18 @@ Vagrant.configure("2") do |config|
         ansible.extra_vars = {
           loopback_ip: properties[:ipAddrPrefix] + iter.to_s
         }
-        ansible.playbook = "clospf.yml"
+        ansible.playbook = "provisioning/clospf.yml"
       end
+      
+      node.vm.provision "shell" do |s|
```
Comment (Vanders) [inline vagrant/demos/clos-ospf/Vagrantfile]: Same comment as above

File: vagrant/demos/clos-ospf/Vagrantfile (hunk 1 above)
Comment (beddari) [inline vagrant/demos/clos-ospf/Vagrantfile]: Hooking into vagrant ssh-config is an option :)

**Comment by Vanders** [comment]
This looks largely good to me; using cl_interfaces is great. We can deal with the addition of the Cumulus Ansible modules using Librarian (I'll email you with details)

File: vagrant/demos/clos-ospf/Vagrantfile (hunk 1 above)
Comment (slaffer-au) [inline vagrant/demos/clos-ospf/Vagrantfile]: This was done to show a consistent work-flow with real hardware in conjunction with https://github.com/slaffer-au/vagrant-to-local. "vagrant ssh" still works with this configuration.

Comment (beddari) [inline vagrant/demos/clos-ospf/Vagrantfile]: @slaffer-au Interesting. We're flowing work in the other direction, e.g connecting physical to vagrant, using https://github.com/tknerr/vagrant-managed-servers :-)

**Comment by slaffer-au** [comment]
@Vanders , having an Ansiblefile appeared to have no effect. Checkout 33597cc and let me know what you think.

**Comment by LeslieCarr** [comment]
+1 for me.  Get a +1 from someone who is more ansible-savvy and then we will merge ! @skamithi @Vanders

**Pull request closed without merging**